shared/src/vault.rs

```diff
@@ -0,0 +1,65 @@
+//! Module containing utilities for interfacing with the Balancer V2 Vault contract.
+
+use anyhow::Result;
+use contracts::BalancerV2Authorizer;
+use ethcontract::{web3::signing, Bytes, H160};
+
+mod sighash {
+    use hex_literal::hex;
+
+    pub const MANAGE_USER_BALANCE: [u8; 4] = hex!("0e8e3e84");
+    pub const BATCH_SWAP: [u8; 4] = hex!("945bcec9");
+}
+
+fn role_id(target: H160, signature: [u8; 4]) -> Bytes<[u8; 32]> {
+    let mut data = [0u8; 36];
+    data[12..32].copy_from_slice(&target.0);
+    data[32..36].copy_from_slice(&signature);
+    Bytes(signing::keccak256(&data))
+}
+
+pub async fn grant_required_roles(
+    authorizer: BalancerV2Authorizer,
+    vault: H160,
+    vault_relayer: H160,
+) -> Result<()> {
```

Comment:
I was under the impression that this was a one time thing... Do we really need code for it?

Reply:
We need this for the unit test in `account_balances.rs` and for E2E tests. From the PR description:

> This is intended to be used in E2E tests in a future PR.